mm/2s2h/Rando/Logic/GlitchlessLogic.cpp

```diff
@@ -96,7 +96,13 @@ void ApplyGlitchlessLogicToSaveContext(std::unordered_map<RandoCheckId, bool>& c
 
                     RandoItemId randoItemId;
 
-                    if (isShuffled) {
+                    if (RANDO_SAVE_CHECKS[randoCheckId].skipped) {
+                        randoItemId = RI_JUNK;
+                        auto findIt = std::find(itemPool.begin(), itemPool.end(), RI_JUNK);
+                        if (findIt != itemPool.end()) {
+                            itemPool.erase(findIt);
+                        }
```

Comment:
```suggestion
                        for (auto it = itemPool.begin(); it != itemPool.end(); ++it) {
                            if (it != itemPool.end()) {
                                if (Rando::StaticData::Items[*it].randoItemType == RITYPE_JUNK) {
                                    randoItemId = *it;
                                    itemPool.erase(it);
                                    break;
                                }
                            }
                        }
```

Because this is looking specifically for RI_JUNK, seeds are likely to fail to generate without many shuffle options selected. The base check pool only has one instance of RI_JUNK for example. This suggestion will allow any item with RITYPE_JUNK to be selected, allowing for more checks to be excluded.